sdk/digitaltwins/digital-twins-parser/review/digital-twins-parser.api.md

```diff
@@ -0,0 +1,507 @@
+    // (undocumented)
+    enumValue?: any;
+}
+
+// @public (undocumented)
+export enum ExtensionKind {
+    // (undocumented)
+    ADJUNCTTYPE = 5,
+    // (undocumented)
+    LATENTTYPE = 6,
+    // (undocumented)
+    NAMEDLATENTTYPE = 7,
+    // (undocumented)
+    NONE = 0,
+    // (undocumented)
+    SEMANTICTYPE = 1,
+    // (undocumented)
+    SEMANTICUNIT = 2,
+    // (undocumented)
+    UNIT = 3,
+    // (undocumented)
+    UNITATTRIBUTE = 4
+}
+
+// @public (undocumented)
+export interface FieldInfo extends SchemaFieldInfo {
+    // (undocumented)
+    entityKind: "field";
+}
+
+// @public (undocumented)
+export interface FloatInfo extends NumericSchemaInfo {
+    // (undocumented)
+    entityKind: "float";
+}
+
+// @public (undocumented)
+export interface IntegerInfo extends NumericSchemaInfo {
+    // (undocumented)
+    entityKind: "integer";
+}
+
+// @public (undocumented)
+export interface InterfaceInfo extends EntityInfo {
+    // (undocumented)
+    contents?: {
+        [value: string]: ContentInfo;
+    };
+    // (undocumented)
+    entityKind: "interface";
+    // (undocumented)
+    extends?: InterfaceInfo[];
+    // (undocumented)
+    schemas?: ComplexSchemaInfo[];
+}
+
+// @public (undocumented)
+export type LanguageStringType = {
+    [languageCode: string]: string;
+};
+
+// @public (undocumented)
+export interface LatentTypeInfo extends EntityInfo {
+    // (undocumented)
+    entityKind: "latenttype";
+}
+
+// @public (undocumented)
+export interface LongInfo extends NumericSchemaInfo {
+    // (undocumented)
+    entityKind: "long";
+}
+
+// @public (undocumented)
+export interface MapInfo extends ComplexSchemaInfo {
+    // (undocumented)
+    entityKind: "map";
+    // (undocumented)
+    mapKey?: MapKeyInfo;
+    // (undocumented)
+    mapValue?: MapValueInfo;
+}
+
+// @public (undocumented)
+export interface MapKeyInfo extends NamedEntityInfo {
+    // (undocumented)
+    entityKind: "mapkey";
+    // (undocumented)
+    schema?: StringInfo;
+}
+
+// @public (undocumented)
+export interface MapValueInfo extends SchemaFieldInfo {
+    // (undocumented)
+    entityKind: "mapvalue";
+}
+
+// @public
+export class MaterialTypeNameCollection {
+    // (undocumented)
+    static initialize(): void;
+    // (undocumented)
+    static isMaterialType(typeString: string): boolean;
+    // (undocumented)
+    static typeNames: Set<string>;
+}
+
+// @public (undocumented)
+export type ModelDict = {
+    [id: string]: EntityInfo | undefined;
+};
+
+// @public (undocumented)
+export interface ModelParser {
+    // (undocumented)
+    dtmiResolver?: DtmiResolver;
+    // (undocumented)
+    getSupplementalTypeCollection(): SupplementalTypeCollection;
+    // (undocumented)
+    maxDtdlVersion?: number;
+    // (undocumented)
+    options: ModelParsingOption;
+    // (undocumented)
+    parse(jsonTexts: string[]): Promise<ModelDict>;
+}
+
+// @public (undocumented)
+export enum ModelParsingOption {
```

Comment:
I am assuming these get used as bit flags hence the powers of 2? Typically we avoid this in JS and prefer options bags e.g.

```ts
{
   parseAllowsIdReferenceSyntax?: boolean;
   permitAnyTopLevelElement?: boolean;
   // and so on...
}
```